mlir-sycl/test/Transforms/constant-propagation.mlir

```diff
@@ -1,42 +1,1155 @@
-  gpu.func @k0(%res: memref<1xi64>, %x: i64, %c: i64, %y: i64) kernel {
-    %add = arith.addi %x, %c : i64
-    %mul = arith.muli %add, %y : i64
-    affine.store %mul, %res[0] : memref<1xi64>
+  gpu.func @k0(%res: memref<1xi64>, %c: i64) kernel {
+    affine.store %c, %res[0] : memref<1xi64>
     gpu.return
   }
 }
 
-// CHECK-LABEL:   llvm.func internal @foo(
-// CHECK-SAME:                            %[[VAL_0:.*]]: !llvm.ptr,
-// CHECK-SAME:                            %[[VAL_1:.*]]: i64,
-// CHECK-SAME:                            %[[VAL_2:.*]]: i64) {
-// CHECK-NEXT:      %[[C:.*]] = llvm.mlir.constant(0 : i64) : i64
-// CHECK-NEXT:      sycl.host.schedule_kernel @kernels::@k0(%[[VAL_0]], %[[VAL_1]], %[[C]], %[[VAL_2]]) : (!llvm.ptr, i64, i64, i64) -> ()
-// CHECK-NEXT:      llvm.return
-// CHECK-NEXT:    }
-llvm.func internal @foo(%res: !llvm.ptr, %x: i64, %y: i64) {
+llvm.func internal @foo(%res: !llvm.ptr) {
   %c = llvm.mlir.constant(0 : i64) : i64
-  sycl.host.schedule_kernel @kernels::@k0(%res, %x, %c, %y)
+  sycl.host.schedule_kernel @kernels::@k0(%res, %c) : (!llvm.ptr, i64) -> ()
+  llvm.return
+}
+
+// -----
+
+!sycl_array_1_ = !sycl.array<[1], (memref<1xi64>)>
+!sycl_id_1_ = !sycl.id<[1], (!sycl_array_1_)>
+!sycl_range_1_ = !sycl.range<[1], (!sycl_array_1_)>
+!sycl_array_2_ = !sycl.array<[2], (memref<2xi64>)>
+!sycl_id_2_ = !sycl.id<[2], (!sycl_array_2_)>
+!sycl_range_2_ = !sycl.range<[2], (!sycl_array_2_)>
+!sycl_array_3_ = !sycl.array<[3], (memref<3xi64>)>
+!sycl_id_3_ = !sycl.id<[3], (!sycl_array_3_)>
+!sycl_range_3_ = !sycl.range<[3], (!sycl_array_3_)>
+!sycl_nd_range_3_ =
+    !sycl.nd_range<[3], (!sycl_range_3_, !sycl_range_3_, !sycl_id_3_)>
+
+gpu.module @kernels {
+// CHECK-LABEL:     func.func @__init.1_k0_impl.1(
+// CHECK-SAME:                                    %[[VAL_0:.*]]: memref<1x!sycl_id_1_>, %[[VAL_1:.*]]: memref<1x!sycl_range_1_>, %[[VAL_2:.*]]: memref<1x!sycl_range_1_>, %[[VAL_3:.*]]: memref<1x!sycl_range_1_>) {
+// CHECK-NEXT:        %[[VAL_4:.*]] = arith.constant 0 : index
+// CHECK-NEXT:        %[[VAL_5:.*]] = sycl.id.constructor() : () -> memref<1x!sycl_id_1_>
+// CHECK-NEXT:        %[[VAL_6:.*]] = memref.load %[[VAL_5]]{{\[}}%[[VAL_4]]] : memref<1x!sycl_id_1_>
+// CHECK-NEXT:        %[[VAL_7:.*]] = arith.constant 0 : index
+// CHECK-NEXT:        %[[VAL_9:.*]] = sycl.num_work_items : !sycl_range_1_
+// CHECK-NEXT:        %[[VAL_10:.*]] = sycl.work_group_size : !sycl_range_1_
+// CHECK-NEXT:        %[[VAL_11:.*]] = sycl.num_work_groups : !sycl_range_1_
+// CHECK-NEXT:        memref.store %[[VAL_6]], %[[VAL_0]]{{\[}}%[[VAL_7]]] : memref<1x!sycl_id_1_>
+// CHECK-NEXT:        memref.store %[[VAL_9]], %[[VAL_1]]{{\[}}%[[VAL_7]]] : memref<1x!sycl_range_1_>
+// CHECK-NEXT:        memref.store %[[VAL_10]], %[[VAL_2]]{{\[}}%[[VAL_7]]] : memref<1x!sycl_range_1_>
+// CHECK-NEXT:        memref.store %[[VAL_11]], %[[VAL_3]]{{\[}}%[[VAL_7]]] : memref<1x!sycl_range_1_>
+// CHECK-NEXT:        return
+// CHECK-NEXT:      }
```

Comment:
We create a constant `id` with `sycl.id.constructor()` and store it in `VAL_0` instead of using `sycl.global_offset`.

Similar changes will be made to other functions.

Reply:
Got it, thanks!

Reply:
Could be useful to add that you mentioned as a comment in the test case?